activerecord/lib/active_record/encryption/scheme.rb

```diff
@@ -45,10 +45,7 @@ def fixed?
       end
 
       def key_provider
-        @key_provider ||= begin
-          validate_keys!
-          @key_provider_param || build_key_provider
-        end
+        @key_provider ||= (@key_provider_param || key_provider_from_config || build_key_provider)
```

Comment:
I think the order here should be 

```ruby
@key_provider_param || build_key_provider || key_provider_from_config
```

If passing an inline `key` param you want it to take priority over the global key provider.